src/Tizen.Nlp/Tizen.Nlp/NaturalLanguageProcess.cs

```diff
@@ -0,0 +1,298 @@
+                result.Add("tag", (string[])msg.GetItem("return_tag"));
+                e.Message = result;
+                _namedentityrecognitionasyncasync?.Invoke(e);
+            }
+            else if (msg.GetItem("command").Equals("lemmatize"))
+            {
+                result.Add("token", (string[])msg.GetItem("return_token"));
+                e.Message = result;
+                _Lemmatizeasync?.Invoke(e);
+            }
+            else if (msg.GetItem("command").Equals("langdetect"))
+            {
+                result.Add("token", (string[])msg.GetItem("return_token"));
+                e.Message = result;
+                _langdetectasync.Invoke(e);
+            }
+            else
+            {
+                return;
+            }
+            Log.Debug(_tag, "done");
+        }
+
+        /// <summary>
+        /// An construct method  to connect remote tidl service.
+        /// </summary>
+        /// <since_tizen> 5 </since_tizen>
+        public NaturalLanguageProcess()
+        {
+            _tag = Application.Current.ApplicationInfo.ApplicationId;
+            Log.Debug(_tag, "msg construct started");
+            _msg = new Message(ServiceId);
+            Log.Debug(_tag, "msg construct success");
+            _noti.Received += ResultReceived;
+            Log.Debug(_tag, "notify callback be assigned");
+            _msg.Connected += (sender, e) =>
+            {
+                Log.Debug(_tag, "start to register");
+                _msg.CoRegister(Application.Current.ApplicationInfo.ApplicationId, _noti);
+                Log.Debug(_tag, "connected callback be called");
+            };
+            _msg.Rejected += (sender, e) =>
+            {
+                Log.Debug(_tag, "rejected callback be called");
+            };
+            _msg.Disconnected += (sender, e) =>
+            {
+                Log.Debug(_tag, "disconnected callback be called");
+            };
+            Log.Debug(_tag, "start to connect");
+            _msg.Connect();
+            Log.Debug(_tag, "wait to callback of onConnected");
+        }
+
+        /// <summary>
+        /// An release session to disconnect remote tidl service.
+        /// </summary>
+        /// <since_tizen> 5 </since_tizen>
+        public void Release()
+        {
+            _noti.Received -= ResultReceived;
+            _msg.UnRegister();
+            _msg.Dispose();
+            _msg = null;
+
+        }
+
+        /// <summary>
+        /// Send Pos of Tag request to remote tidl service with one parameters.
+        /// </summary>
+        /// <param name="sentence">A sentence need to be processed.</param>
+        /// <returns>PosTagResult</returns>
+        /// <since_tizen> 5 </since_tizen>
+        public Task<PosTagResult> PosTagAsyncTask(string sentence)
+        {
+            MakeRequest("pos_tag", sentence);
+            var task = new TaskCompletionSource<PosTagResult>();
+            _postagasync = (e) =>
```

Comment:
When 2 requests was sent, It will not delivered the result to first request. so you need to distinguish each request using identifier like a transaction id

```c#
var first = PosTagAsyncTask("....");
var second = PosTagAsyncTask(".....");
var result = await first; // it will never returned. 
```

Reply:
You need to more study on `Task` and asynchronous
```c#
 PostTagAsyncTask("...").ContinueWith((t)=>{ someAction(t.Result); }
 PostTagAsyncTask("....2").ContinueWith((t)=>{ someAction(t.Result); }
```